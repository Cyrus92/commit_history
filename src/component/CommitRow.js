import React from 'react';

export default class CommitRow extends React.Component {
  constructor(props) {
    super(props);
    
}
  render() {
    // console.log( this.props.githubCommitItem)
    return (
      <div className='row'>
        <div className='col-md-12 commitItem'>
          <h6 className='commitItemLink text-left'><a target="_blank" href={this.props.githubCommitItem?this.props.githubCommitItem.html_url:"#"}>{this.props.githubCommitItem?this.props.githubCommitItem.commit.message:""}</a></h6>
          <div className=''>
              <img className='commitItemImage float-left' src={this.props.githubCommitItem?this.props.githubCommitItem.author.avatar_url:""} />
              <p className='commitItemAuthor text-left'>{this.props.githubCommitItem?this.props.githubCommitItem.commit.author.name:""} committed on&nbsp;
              {
                this.props.githubCommitItem?
                new Date(this.props.githubCommitItem.commit.author.date ).toLocaleDateString():""
              }
              </p>
          </div>
        </div>

      </div>

    )
  }
}