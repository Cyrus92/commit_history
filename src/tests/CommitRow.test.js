import React from 'react';
import { shallow, mount } from 'enzyme';
import CommitRow from '../component/CommitRow';
describe("CommitRow component", () => {

    it("should render", () => {
        const wrapper = shallow(<CommitRow />);
    });

});