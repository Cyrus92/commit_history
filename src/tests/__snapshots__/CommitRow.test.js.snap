// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommitRow component should render initial layout 1`] = `
Array [
  <div
    className="row"
  >
    <div
      className="col-md-12 commitItem"
    >
      <h6
        className="commitItemLink text-left"
      >
        <a
          href="#"
          target="_blank"
        >
          
        </a>
      </h6>
      <div
        className=""
      >
        <img
          className="commitItemImage float-left"
          src=""
        />
        <p
          className="commitItemAuthor text-left"
        >
          
           committed onÂ 
          
        </p>
      </div>
    </div>
  </div>,
]
`;
